packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -168,10 +187,41 @@ mixin RenderConstrainedLayoutBuilder<ConstraintType extends Constraints, ChildTy
     markNeedsLayout();
   }
 
-  /// Invoke the layout callback.
-  void layoutAndBuildChild() {
+  bool _needsBuild = true;
+
+  /// Marks this layout builder as needing to rebuild.
+  ///
+  /// The layout build rebuilds automatically when layout constraints change.
+  /// However, we must also rebuild when the widget updates, e.g. after
+  /// `setState`, or `didChangeDependencies`, even when the layout constraints
```

Comment:
use `[State.setState]` et al rather than \`\` so that people can follow cross-refs.

Reply:
Done.